components/files.js

```diff
@@ -6,127 +6,297 @@ const firstBy = require('thenby');
+              return value;
+            },
+          }))
+          .on('data', (data) => {
+            // if prune is set and we are sorting, we have to ensure
+            // the rows have all the columns to sort on
+            if ('prune' in this.config && 's' in this.config) {
+              const curKz = Object.keys(data);
+              const sortOn = this.config.s.reduce((r, k) => r.concat(k.column), []);
+              if (sortOn.every(itm => curKz.includes(itm))) {
+                common.cache[this.universe].push(data);
+              }
+            } else {
+              common.cache[this.universe].push(data);
             }
-            return value;
-          },
-        }))
-        .on('data', (data) => { common.cache[universe].push(data); })
-        .on('error', () => reject())
-        .on('end', () => resolve(_filterData()));
-    }).then(response => response);
+          })
+          .on('error', () => reject())
+          .on('end', () => {
+            this.total_data = common.cache[this.universe].length;
+            resolve(this.filterData());
+          });
+      }).then(response => response);
+    }
   }
-};
 
-/**
- * Stub - Only "publicly" available method for getting data
- * @param  {String} unvrs Comic Universe (DC or Marvel)
- * @param  {Object} cfg   Configuration Object
- * @return {Promise}      Promise object of results
- */
-const readFile = (unvrs, cfg) => _getData(unvrs, cfg);
+  /**
+   * Stub - Only "publicly" available method for getting data
+   * @param  {Object} config   Configuration Object
+   * @return {Promise}      Promise object of results
+   */
+  readCharacterFile(config) {
+    this.config = config;
+    return this.getData();
+  }
+
+  /**
+   * Custom sorting function that allows for sorting None while also maintaining
+   * locale aware sorting config["nulls"] will allow for sorting None and putting
+   * them at the front or end of the list
+   * l = [1, 3, 2, 5, 4, null, 7]
+   * console.log('Last = ', sort_results(l)) e.g nulls=last
+   * Last = [1, 2, 3, 4, 5, 7, None]
+   * console.log('First = ', sort_results(l)) e.g nulls=first
+   * First = [None, 1, 2, 3, 4, 5, 7]
+   * @param  {Array} results Array of characters leftover from other filters
+   * @return {Array} Sorted array of JSON objects based on criteria
+   */
+  sortResults(results) {
+    // https://github.com/Teun/thenBy.js
+    // The first entry in the array will always the the first solumn we sort on
+    let srt = firstBy(this.config.s[0].column,
+      { cmp: this.sortIl8nStr.bind(this), direction: this.config.s[0].sort });
+    // If there are more than one item then we dynamically sort on those
+    for (let i = 1; i < this.config.s.length; i += 1) {
+      srt = srt.thenBy(this.config.s[i].column,
+        { cmp: this.sortIl8nStr.bind(this), direction: this.config.s[i].sort });
+    }
+    return results.sort(srt);
+  }
+
+  /**
+   * Sorting compare function to handle
```

Comment:
The functionality here is explained in the method above. Does it make sense to detail it here instead?

Reply:
i think it's fine in the above method